### ISSUE TITLE
Need a 3-point ellipse constructor

### ISSUE THREAD

**Issue by dassaf4** [iTwin/itwinjs-core#6794]
Booster needs to construct an ellipse (`Arc3d`) through 3 points like MicroStation does.

Though 3 general points define a circle (cf. `Arc3d.createCircularStartMiddleEnd`), this is not enough information to specify a non-circular ellipse. However, if we constrain two of the points to lie on opposite ends of an axis of the ellipse, the construction is possible.

**Comment by dassaf4** [comment]
Probably should backport this to native, as the construction does not exist there.

- [ ] backport to imodel-native
- [ ] backport to imodel02
- [ ] backport to PPBase

**Comment by saeeedtorabi** [comment]
> if we constrain two of the points to lie on opposite ends of an axis of the ellipse, the construction is possible.

1- How can we force the constrain? Would you please provide more info on this?

2- TS code only supports circle. I assume we need to port back ellipse construction from native to TS, right?

**Comment by dassaf4** [comment]
Yes, this issue remains open for backport of the new iTwin method. I added a new "Needs backport" status for issues like this, so I've updated its status. The 3pt ellipse construction assumes first and third points are on an axis, and middle point is somewhere else on the arc. If the 3 points are all equidistant from another point, a circle is created, but this just falls out of the math; we don't special case it. In general, if you want to create a circle, it's more convenient to use other methods. But it would be a good test to add: given 3 points on a circle, with first and last on a diameter, both 3pt methods should yield the same circle (geometrically if not parametrically)!

**Comment by saeeedtorabi** [comment]
1- I thought more about ellipse with 3pt. Isnt it super easy? If you have points A, B, and C connect a line from A to C and find the middle point on that line (call it M). Then M would be ellipse center, MC would be vector0, and MB vector90.

2- I'm confused! Do we need to port `Arc3d.createCircularStartMiddleEnd` from TS to native or not?  Isnt `bsiDEllipse3d_constructFillet` in `dellipse3d.cpp` that method?

3- Do we need a new method to create an arc using 3pt in native (and TS)? What is `bsiDEllipse3d_initFrom3DPoint3dOnArc` in `dellipse3d.cpp` then?

**Comment by saeeedtorabi** [comment]
imodel-native PR: https://github.com/iTwin/imodel-native/pull/941

_I will wait for this to be merged. Then I will port it to PPBase_

**Comment by saeeedtorabi** [comment]
PPBase PBI: https://dev.azure.com/bentleycs/PowerPlatform/_workitems/edit/1588678

PPBase PR: https://dev.azure.com/bentleycs/PowerPlatform/_git/PPBase/pullrequest/491036

**Comment by saeeedtorabi** [comment]
All PR are merged.
